### DISCUSSION TITLE
Sharing Eslint Config: Parsing error: Cannot read file `tsconfig.json`

### DISCUSSION THREAD

**Discussion by arkmech** [vercel/turborepo#3342] [Help] [answered]
Basing this off mostly on the `with-tailwind` example.

Error I'm getting in all my files. `Parsing error: Cannot read file 'tsconfig.json'.`

It properly lints my files, but on all my files have the above error. I have researched other posts to use `__dirname`, but then it doesn't lint my files, which is why I use `"."` instead for `tsconfigRootDir`.

`eslint-config-custom` `index.js` for `["*.ts", "*.tsx"]`
```
overrides: [
    {
      files: ["*.ts", "*.tsx"],
      extends: [
        "airbnb-typescript",
        "plugin:@typescript-eslint/recommended",
        "plugin:prettier/recommended",
      ],
      parser: "@typescript-eslint/parser",
      parserOptions: {
        sourceType: "module",
        project: "./tsconfig.json",
        tsconfigRootDir: ".",
      },
    }
]
```

`eslint-config-custom` `package.json`
```json
  "dependencies": {
    "@typescript-eslint/eslint-plugin": "^5.48.2",
    "@typescript-eslint/parser": "^5.48.2",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-airbnb-typescript": "^17.0.0",
    "eslint-config-next": "latest",
    "eslint-config-prettier": "^8.5.0",
    "eslint-config-turbo": "latest",
    "eslint-plugin-import": "^2.27.5",
    "eslint-plugin-jest": "^27.2.1",
    "eslint-plugin-jest-dom": "^4.0.3",
    "eslint-plugin-jsx-a11y": "^6.7.1",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-react": "7.28.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-tailwindcss": "^3.8.0",
    "eslint-plugin-testing-library": "^5.9.1"
  },
  "devDependencies": {
    "tsconfig": "workspace:*",
    "typescript": "^4.9.4"
  },
```

Where should I keep the `.vscode` `settings.json` so that I can lint on save?
A: Within each project
B: Root of TurboRepo
```
{
  "editor.codeActionsOnSave": { "source.fixAll.eslint": true },
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnSave": true
}
```

**Comment by anthonyshew** [comment, marked as answer]
We've recently updated the examples such that they all should work out of the box with `typescript-eslint`

Check them out here: https://github.com/vercel/turbo/tree/main/examples

To address your second question, `.vscode` folders can be nested into directory tree. Usually, you'll see one in the monorepo root to handle everything.
